packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -280,34 +283,43 @@ class RequestManager {
 
         await _cache?.set(key, serialized, cacheParameters);
         break;
-      } on TimeoutException catch (e, s) {
-        debugPrint('Request timed out ...');
-        debugPrint(e.toString());
-        debugPrintStack(stackTrace: s, maxFrames: 5);
+      } on TimeoutException catch (error, stackTrace) {
+        _log.info(
+          'Request timeout',
+          error,
+          stackTrace,
+        );
 
         subject.add(
           subject.value.copyWith(
-            error: e,
+            error: error,
             isLoading: false,
           ),
         );
         break;
-      } on http.ClientException catch (e, s) {
-        debugPrint(e.toString());
-        debugPrintStack(stackTrace: s, maxFrames: 5);
+      } on http.ClientException catch (error, stackTrace) {
+        _log.info(
+          'Error while executing the request. Retrying ...',
+          error,
+          stackTrace,
+        );
+
+        if (error is! DynamiteStatusCodeException || error.statusCode < 500) {
+          _log.warning(
+            'Unexpected status code. The request will not be retried.',
+            error,
+            stackTrace,
+          );
 
-        if (e is! DynamiteStatusCodeException || e.statusCode < 500) {
           subject.add(
             subject.value.copyWith(
-              error: e,
+              error: error,
               isLoading: false,
             ),
           );
           break;
         }
       }
-
-      debugPrint('Retrying...');
```

Comment:
This message should stay here because it is the only place we reliably know that we are going to retry without logging in multiple places.